examples/bigquery-analyze-realtime-reddit-data/scripts/setup.sh

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Run in Cloud Shell to set up your project and deploy solution.
+
+echo "===================================================="
+echo " Setting up environment variables ..."
+
+usage() {
+    echo "Usage: [ -i projectId ] [ -r region ] [ -c redditClientId ] [ -u redditUser ]"
+}
+export -f usage
+
+while getopts ":i:r:c:u:" opt; do
+    case $opt in
+        i ) projectId="$OPTARG";;
+        r ) region="$OPTARG";;
+		c ) redditClientId="$OPTARG";;
+		u ) redditUser="$OPTARG";;
+        \?) echo "Invalid option -$OPTARG"
+        usage
+        exit 1
+        ;;
+    esac
+done
+
+
+echo "===================================================="
+echo " Inputs ..."
+echo " Project ID: ${projectId}" 
+echo " Region: ${region}" 
+echo " Reddit Client Id: ${redditClientId}"
+echo " Reddit User: ${redditUser}" 
+
+
+# GCP
+export APP_BUCKET=${projectId}-reddit-stream-app
+
+# reddit
+echo "Please enter reddit client secret:"
+read -s -r REDDIT_CLIENT_SECRET
+echo "Please enter reddit password:"
+read -s -r REDDIT_PASSWORD
+
+echo "===================================================="
+echo " Enabling services ..."
+
+gcloud config set project "$projectId"
+
+gcloud services enable storage-component.googleapis.com 
```

Comment:
have you considered using https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/google_project_service instead?

A nice pattern for that is:
```
locals {
  apis = ["iam.googleapis.com", "run.googleapis.com", "apigateway.googleapis.com", "servicecontrol.googleapis.com", "servicemanagement.googleapis.com", "cloudapis.googleapis.com", "cloudresourcemanager.googleapis.com", "identitytoolkit.googleapis.com", "secretmanager.googleapis.com", "compute.googleapis.com", "iap.googleapis.com"]
}

data "google_project" "project" {
  project_id = var.project_id
}

resource "google_project_service" "project" {
  for_each = toset(local.apis)
  project  = data.google_project.project.project_id
  service  = each.key

  disable_on_destroy = false
}
```

This way you can annotate the resources in Terraform that depend on the API that needs to be enabled with depends on:
```
 depends_on = [
    google_project_service.project
  ]
```